packages/plugin/src/Jobs/SendNotificationsJob.php

```diff
@@ -88,8 +90,17 @@ public function execute($queue): void
         // Set the application language to the site's primary language
         \Craft::$app->language = $sites->getCurrentSite()->language;
 
-        $submission = $freeform->submissions->getSubmissionById($this->submissionId);
-        if (!$submission || $submission->isSpam) {
+        $submission = null;
+        if ($this->submissionId) {
+            $submission = $freeform->submissions->getSubmissionById($this->submissionId);
+        }
+
+        // If no stored submission and we have posted data, fabricate one in-memory
+        if (!$submission && $this->postedData) {
+            $submission = $this->createTmpSubmission($form);
+        }
```

Comment:
Even if posted data could be empty, it should still be present every time, so we can always create a new submission in the case of missing submission ID

```suggestion
        $submission = null;
        if ($this->submissionId) {
            $submission = $freeform->submissions->getSubmissionById($this->submissionId);
        } else {
            // If no stored submission - fabricate one in-memory
            $submission = $this->createTmpSubmission($form);
        }
```